docs/sources/operations/request-validation-rate-limits.md

```diff
@@ -0,0 +1,226 @@
+| Enforced by             | `distributor`     |
+| Outcome                 | Request rejected  |
+| Retryable               | **No**            |
+| Sample discarded        | **Yes**           |
+| Configurable per tenant | Yes               |
+| HTTP status code        | `400 Bad Request` |
+
+## `too_far_in_future`
+
+If a sample's timestamp is greater than the current timestamp, Loki allows for a certain grace period during which samples will be accepted. If the grace period is exceeded, the error will occur.
+
+This value can be modified globally in the [`limits_config`](https://grafana.com/docs/loki/latest/configuration/#limits_config) block, or on a per-tenant basis in the [runtime overrides](https://grafana.com/docs/loki/latest/configuration/#runtime-configuration-file) file. This error can be solved by increasing the `creation_grace_period` value, or investigating why this particular stream has a timestamp too far into the future. The stream in question will be returned in the body of the HTTP response.
+
+| Property                | Value             |
+|-------------------------|-------------------|
+| Enforced by             | `distributor`     |
+| Outcome                 | Request rejected  |
+| Retryable               | **No**            |
+| Sample discarded        | **Yes**           |
+| Configurable per tenant | Yes               |
+| HTTP status code        | `400 Bad Request` |
+
+## `max_label_names_per_series`
+
+If a series is submitted with more labels than Loki has been configured to allow, this error will occur.
+
+This value can be modified globally in the [`limits_config`](https://grafana.com/docs/loki/latest/configuration/#limits_config) block, or on a per-tenant basis in the [runtime overrides](https://grafana.com/docs/loki/latest/configuration/#runtime-configuration-file) file. This error can be solved by increasing the `max_label_names_per_series` value. The offending stream will be returned in the body of the HTTP response.
+
+| Property                | Value             |
+|-------------------------|-------------------|
+| Enforced by             | `distributor`     |
+| Outcome                 | Request rejected  |
+| Retryable               | **No**            |
+| Sample discarded        | **Yes**           |
+| Configurable per tenant | Yes               |
+| HTTP status code        | `400 Bad Request` |
+
+## `label_name_too_long`
+
+If a series has a label name with a length in bytes greater than Loki has been configured to allow, this error will occur.
+
+This value can be modified globally in the [`limits_config`](https://grafana.com/docs/loki/latest/configuration/#limits_config) block, or on a per-tenant basis in the [runtime overrides](https://grafana.com/docs/loki/latest/configuration/#runtime-configuration-file) file. This error can be solved by increasing the `max_label_name_length` value. The offending stream will be returned in the body of the HTTP response.
```

Comment:
in the future would be nice to return not only the stream but also make clear which is the specific label name in that stream that's the source of the problem.

Reply:
do we have suggestions for users on how high they should go with this? like don't set `max_label_name_length` above x?

Reply:
> in the future would be nice to return not only the stream but also make clear which is the specific label name in that stream that's the source of the problem.

Agreed.

> do we have suggestions for users on how high they should go with this? like don't set max_label_name_length above x?

I'm not aware of any specific limit, but it's `1024` by default so I'd say that's already excessively high.